devices.go

```diff
@@ -172,10 +183,48 @@ func (d *deviceCache) SetDeviceByIdOpState(id string, os models.OperatingState)
 
 // Update updates the device in the cache and ensures that the
 // copy in Core Metadata is also updated.
-func (d *deviceCache) Update(id string) error {
+func (d *deviceCache) Update(dev *models.Device) error {
+	addressable, err := updateAddressable(&dev.Addressable)
+	if err != nil {
+		return err
+	}
+	dev.Addressable = *addressable
+
+	device, err := svc.dc.DeviceForName(dev.Name)
+	if err != nil {
+		return err
+	}
+
+	dev.Id = device.Id
```

Comment:
Why it's needed `svc.dc.DeviceForName()` and then `dev.Id = device.Id`? Device name can change, but device ID shouldn't change, so there is no need to call `svc.dc.DeviceForName()`

Reply:
Thanks, I agree.